### ISSUE TITLE
[Feature] Proposal of the e2e testing of UI

### ISSUE THREAD

**Issue by LinuxSuRen** [apache/skywalking#11574]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/skywalking/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

I found out the e2e testing of [skywalking-booster-ui](https://github.com/apache/skywalking-booster-ui) is empty. I did some UI e2e testing work in [another project](https://github.com/LinuxSuRen/api-testing/tree/master/console/atest-ui/cypress/e2e). So, I'm thinking about if I could complete it in Skywalking UI project.

## How to
Firstly, we could start the backend and the related agent project (for getting the tracing data). Docker-compose could do it in a good manner.

Then, Cypress is a good tool to visit the page and do the necessary assert.

## Potential issues
Considering we need to find the UI element via some pattern. (for instance: ID, css, or others). But it's not easy due to no easy and readable way to do that currently. I know one way to make it be easier. We could just add an identity attribute to the elements. But the code will be ungly in some way. Please let me know if your have any thoughts or good idea.

See also this [example](https://github.com/LinuxSuRen/api-testing/blob/1241aa81e2a13d7429b2d555bf6b96cbc19bac99/console/atest-ui/src/App.vue#L331).

And https://github.com/LinuxSuRen/skywalking-booster-ui/pull/1

### Are you willing to submit a pull request to implement this on your own?

- [X] Yes I am willing to submit a pull request on my own!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by kezhenxu94** [comment]
Tests are always nice to have but adding E2E tests in skywalking-booster-ui doesn't seem to add many benefits in SkyWalking, the SkyWalking UI is mostly for displaying data and has very little user operations, and we already have E2E in OAP to verify the data, if we'd like to add tests in the UI repo, I'd rather add component tests that verify the component renders correctly for given (mocked) data, without having to spin up the entire environment (OAP, agents, storage, etc.), which is more friendly for UI developers.

**Comment by LinuxSuRen** [comment]
I agree with you that the component test seems is valueable than the e2e testing. But I'll try to explore if there is a way to make it be easier to write the UI e2e code.

**Comment by wu-sheng** [comment]
@Fine0830 What do you suggest on UI side test? You mentioned once.

**Comment by Fine0830** [comment]
Glad to see this discussion. It is a good practice to use both unit tests and E2E tests in front-end development. These two types of tests serve different purposes and focus areas, collectively ensuring the quality and stability of a front-end application. Unit(component) tests are primarily used to test the smallest units of the application, such as functions, components, or modules and ensure that individual parts of the code work as expected, helping catch potential logic errors and boundary cases. E2E tests cover the entire application, simulating user interactions with the application.
and ensure that various components and features work together within the entire application context, detecting issues in user scenarios.

These operations can support various behaviors for conducting E2E testing:
1. Simulate users to operate DOM.
2. Listening for API requests and responses.
3. Simulate users clicking on the link to jump to a new tab.

But now I'm not quite sure how to achieve them yet, I'm glad to see this happen if @LinuxSuRen would implement this.
